### PR TITLE
Multidrag modules

### PR DESCRIPTION
**Pull request by saisilinus**
Add support for the selection of multiple modules before dragging
Closes ASKnetCommunity/Training#107

### CODE REVIEW

**Comment by saisilinus** [comment]
Hey @bmen. I  may need your help on this one. Are you familiar with [Sortable](https://sortablejs.github.io/Sortable/)? I've tried implementing the `MultiDrag` feature but I am getting inconsistent results. Sometimes it works, other times it doesn't. I am still trying to figure out what the issue is but another set of eyes wouldn't hurt.

**Comment by saisilinus** [comment]
It should be working now @timmwille, @bmen. I also fixed a couple of errors to do with time calculation. Previously you could get a weird duration like 44477 hours; should be okay now 👍🏿 .

**Comment by bmen** [comment]
At first glance at the code, everything looks fine. I would have to check out your fork and try it out.
Why @saisilinus did you remove multiDragKey? Isn't that the feature you wanted?

Then I thought it might be due to a bug in Sortable, but version 1.15.0 is also already from March 2022, so it's been available for a while.

I also don't have a prompt answer as to why it sometimes works and sometimes doesn't. I'll take a look at your fork and play around a bit. What was the goal? To be able to move several modules in the list?

**Comment by saisilinus** [comment]
> At first glance at the code, everything looks fine. I would have to check out your fork and try it out. Why @saisilinus did you remove multiDragKey? Isn't that the feature you wanted?
> 
> Then I thought it might be due to a bug in Sortable, but version 1.15.0 is also already from March 2022, so it's been available for a while.
> 
> I also don't have a prompt answer as to why it sometimes works and sometimes doesn't. I'll take a look at your fork and play around a bit. What was the goal? To be able to move several modules in the list?

I removed the `multiDragKey` because I wasn't sure that the guys using the app would want that. If the key is set then you have to hold it while selecting multiple items; kind of like you hold `CTRL` when selecting several things on your computer. What do you guys think? @timmwille @bmen

**Comment by saisilinus** [comment]
> At first glance at the code, everything looks fine. I would have to check out your fork and try it out. Why @saisilinus did you remove multiDragKey? Isn't that the feature you wanted?
> 
> Then I thought it might be due to a bug in Sortable, but version 1.15.0 is also already from March 2022, so it's been available for a while.
> 
> I also don't have a prompt answer as to why it sometimes works and sometimes doesn't. I'll take a look at your fork and play around a bit. What was the goal? To be able to move several modules in the list?

The goal was for a user to be able to select several items from the modules list and move them at once. For a proper user experience, users should also be able to deselect items when they have selected them and they should get a visual confirmation that they have selected/deselected a module. It's supposed to fix #107.

I've fixed the issue though so now it should work properly. Let me know if it doesn't work as expected.

**Comment by timmwille** [comment]
looks like it works now, thank you guys it is beautiful

**Comment by bmen** [comment]
Was this feature of adding several modules to the training plan at the same time so important?
But it works.

The operating concept could be reconsidered. I don't know whether so many people understand that you can click on a small circle behind a semi-transparent hover layer to select several modules for drag and drop? I think the small circle with a tick is a smart implementation.

**Comment by bmen** [comment]
Wouldn't it be better if we added a small Add button to each module in the module list? The button inserts the module right at the end of the training plan. This would also allow you to quickly add several modules to the training plan and we would also have a solution for mobile interfaces, as drag & drop does not work so well on mobile devices.

I apologise for coming so late with my suggestion.

**Comment by timmwille** [comment]
> Wouldn't it be better if we added a small Add button to each module in the module list? The button inserts the module right at the end of the training plan. This would also allow you to quickly add several modules to the training plan and we would also have a solution for mobile interfaces, as drag & drop does not work so well on mobile devices.
> 


Thanks for thinking ahead. I think this comment is worth (re)considering when we move into #5 ? and we might change it a bit again. For now I would merge it in as a new feature that really speeds up the time for testing scenarios with the Training planner... I will also see what feedback comes in from @walthierer and others when using the platform.

**Comment by timmwille** [comment]
@saisilinus the conflicts coming from #126 also affect this one, so should be possible to solve in one go?

**Comment by saisilinus** [comment]
> @saisilinus the conflicts coming from #126 also affect this one, so should be possible to solve in one go?

Maybe. Try merging #126 first

**Comment by saisilinus** [comment]
> Was this feature of adding several modules to the training plan at the same time so important? But it works.
> 
> The operating concept could be reconsidered. I don't know whether so many people understand that you can click on a small circle behind a semi-transparent hover layer to select several modules for drag and drop? Nevertheless, I think the small circle with a tick is a smart implementation.

Clicking on the entire module selects it. The circle is just visual feedback

File: assets/js/trainingsplan.js [hunk 1]
```diff
@@ -20,26 +22,35 @@ function initiateSortable() {
...
             put: true
         },
         fallbackOnBody: true,
         swapThreshold: 0.2,
         animation: ANIMATION_SPEED,
+        multiDrag: true,
+        avoidImplicitDeselect: true,
         onAdd: runDynamicCalculationsOnAdd,
-        onUpdate: runDynamicCalculationsOnUpdate
+        onUpdate: runDynamicCalculationsOnUpdate,
```
Comment (timmwille) [inline assets/js/trainingsplan.js:34]: @saisilinus nice catch

**Review by timmwille — APPROVED** [review]
I think the conflicts should no be fixed @saisilinus @bmen - maybe you want to double-check?

**Pull request merged**